src/project-3.js

```diff
@@ -1,83 +1,93 @@
+  cat.age = age;
+  cat.meow = () => 'Meow!';
+  return cat;
 };
 
 const addProperty = (object, property) => {
-  // add the property to the object with a value of null
-  // return the object
-  // note: the property name is NOT 'property'.  The name is the value of the argument called property (a string)
+  object[property] = null;
+  return object;
 };
 
 const invokeMethod = (object, method) => {
-  // method is a string that contains the name of a method on the object
-  // invoke this method
-  // nothing needs to be returned
+  object[method]();
 };
 
 const multiplyMysteryNumberByFive = (mysteryNumberObject) => {
-  // mysteryNumberObject has a property called mysteryNumber
-  // multiply the mysteryNumber property by 5 and return the product
+  return mysteryNumberObject.mysteryNumber * 5;
 };
 
 const deleteProperty = (object, property) => {
-  // remove the property from the object
-  // return the object
+  delete object[property];
+  return object;
 };
 
 const newUser = (name, email, password) => {
-  // create a new object with properties matching the arguments passed in.
-  // return the new object
+  const user = {
+    name,
+    email,
+    password,
+  };
+  return user;
 };
 
 const hasEmail = (user) => {
-  // return true if the user has a value for the property 'email'
-  // otherwise return false
+  if (user.email) {
+    return true;
+  }
+  return false;
 };
 
 const hasProperty = (object, property) => {
-  // return true if the object has the value of the property argument
-  // property is a string
-  // otherwise return false
+  if (object[property]) {
+    return true;
+  }
+  return false;
 };
 
 const verifyPassword = (user, password) => {
-  // check to see if the provided password matches the password property on the user object
-  // return true if they match
-  // otherwise return false
+  if (user.password === password) {
+    return true;
+  }
+  return false;
 };
 
 const updatePassword = (user, newPassword) => {
-  // replace the existing password on the user object with the value of newPassword
-  // return the object
+  user.password = newPassword;
+  return user;
 };
 
 const addFriend = (user, newFriend) => {
-  // user has a property called friends that is an array
-  // add newFriend to the end of the friends array
-  // return the user object
+  user.friends.push(newFriend);
+  return user;
 };
 
 const setUsersToPremium = (users) => {
-  // users is an array of user objects.
-  // each user object has the property 'isPremium'
-  // set each user's isPremium property to true
-  // return the users array
+  for (let i = 0; i < users.length; i++) {
+    users[i].isPremium = true;
+  }
+  return users;
 };
 
 const sumUserPostLikes = (user) => {
-  // user has an array property called 'posts'
-  // posts is an array of post objects
-  // each post object has an integer property called 'likes'
-  // sum together the likes from all the post objects
-  // return the sum
+  let sum = 0;
```

Comment:
This is great.  Now that you're getting more experienced I'd like you to use `reduce` for things like this